src/txmempool.cpp

```diff
@@ -257,9 +260,9 @@ void CTxMemPool::UpdateForRemoveFromMempool(const setEntries &entriesToRemove, b
         // updateDescendants should be true whenever we're not recursively
         // removing a tx and all its descendants, eg when a transaction is
         // confirmed in a block.
-        // Here we only update statistics and not data in mapLinks (which
-        // we need to preserve until we're finished with all operations that
-        // need to traverse the mempool).
+        // Here we only update statistics and not data in CTxMemPool::Parents
+        // and CTxMemPoolEntry::Children (which we need to preserve until we're
```

Comment:
I think what is meaned by "data" is a bit confusing here. It can  be either a _pair_ of iterator/sets in `mapLinks` or the _sets_ in themselves or even _statistics_ of each set element ? Of course, what we care is integrity of links to keep our traversal consistent until we're done updating. But referring now to Parents/Children doesn't make it clearer.

As a better suggestion, comment may explain that our processing 1) update accounting for all interdependent entries of target 2) remove target and pointers/iterators to interdependencies and thus we MUST conserve last ones while doing 1) as otherwise it would fraught our accounting. Where interdependency is defined as a ancestors/descendants relation.

Reply:
Maybe we can clean up the documentation later? I agree this is confusing but it's not made worse by changing it here.